### PR TITLE
Fixes from the second verification sweep (privacy, access, n_num, performance, polish)

### LINKED ISSUE #172: Hall-of-Fame record widgets leak a living holder's derived value + record URL (no canShow gate)
### Summary

The "Hall of fame" record-holder widgets (Overview hero records-grid) resolve and render their holders without a `canShow()` privacy gate. The holder's **name** is privatised by `NameProcessor::getFullName()` (→ "Private" for a living individual a visitor cannot see), but the **derived value** (age / spouse-count / child-count) and the **record URL** (a deep-link carrying the XREF) are rendered unconditionally. For a record held by a *living* individual on a public/guest-visible tree, that discloses a sensitive derived fact plus a navigable handle to the otherwise-private record.

### Locations

- `src/Support/Aggregator/IndividualAgeRecordResolver.php:56-58` — resolves via `Registry::individualFactory()->make($xref)` with only an `instanceof Individual` check, no `canShow()`.
- `src/Repository/ParenthoodRepository.php` — `youngest/oldestParentAtFirstChildRecord` feed candidate xrefs from a raw `families/dates/link` join with no access filter.
- `src/Repository/MarriageRepository.php` — `spouseAtMarriageRecord` (youngest/oldest husband/wife) and `mostSpousesRecord` resolve holders with only `instanceof`.
- `src/Support/Aggregator/RecordRowMapper.php:66,86,106,126,146,166` — `'url' => $record->…->url()` set unconditionally.
- `resources/views/modules/statistics-chart/components/records-grid.phtml:53` — renders `<a href="<?= e($r['url']) ?>">`.

### Contrast (the correct pattern already in the codebase)

- `MarriageRepository.php:404` gates the sensitive `endReason` on `$family->canShow()`.
- `LifeSpanRepository::oldestLivingRecord()` / `topOldestLiving()` are safe because the core `getBirthDate()->isOK()` step skips a living individual whose birth date a visitor cannot see — the Parenthood / Marriage-spouse paths have no equivalent gate.

### Manifestation

Does **not** manifest when the record-holders are deceased (the common case on a well-populated tree — verified). It **does** manifest when a *living* individual holds a superlative (e.g. a living "youngest mother", "most spouses") on a tree whose statistics are visible to guests/visitors — the tile then shows "Private — N years" / "Private — N marriages" plus a clickable link to that individual's record. GDPR-relevant.

### Fix

Gate every resolved record holder on `canShow()`: drop the holder (return `null` / skip the tile) — or at minimum null the derived value AND suppress the `url` — when `!$record->canShow()`. A single guard in the shared resolver covers the Parenthood and Marriage-spouse paths; `mostSpousesRecord` needs its own. Render a `<span>` (no anchor) when `url` is empty.

### Test

Add a privacy-harness test (drop the importing admin to an anonymous visitor, per the project's privacy test pattern) on a fixture where a **living** individual holds a record, asserting the resolved holder is `null` / the rendered tile carries no XREF link and no derived value. The current suite never exercises a `!canShow()` holder, so the gap ships silently.

### LINKED ISSUE #173: AJAX tab actions skip Auth::checkComponentAccess — component access restriction is bypassable
### Summary

The six AJAX tab actions render their data without an `Auth::checkComponentAccess` check. Only the top-level page action (`getChartAction`) enforces the chart component's access level. webtrees' `ModuleAction` dispatcher only verifies the module is enabled (and that `Admin…`-named actions need admin) and explicitly delegates per-component access to the module — so a visitor can call a tab endpoint directly and bypass any access restriction the admin configured for the Statistics chart component.

### Location

`src/Module.php` — `getOverviewAction`, `getNamesAction`, `getLifeSpanAction`, `getFamilyAction`, `getPlacesAction`, `getTreeHealthAction` each call `renderTab(...)` directly with no access check. Only `getChartAction` calls `Auth::checkComponentAccess($this, ModuleChartInterface::class, $tree, $user)`.

### Attack

Admin sets the Statistics chart component access to "members". A guest requests `…?route=/module/<name>/LifeSpan/<tree>` directly; the dispatcher invokes `getLifeSpanAction → renderTab('life-span')` and returns the full tab data, bypassing the restriction (and re-exposing the data the component was meant to hide).

### Fix

Move the access check into the shared `renderTab()` so every tab action enforces it. Thread the `ServerRequestInterface` through each `get*Action` and:

```php
private function renderTab(ServerRequestInterface $request, string $template): ResponseInterface
{
    $tree = Validator::attributes($request)->tree();
    $user = Validator::attributes($request)->user();
    Auth::checkComponentAccess($this, ModuleChartInterface::class, $tree, $user);
    // …existing render…
}
```

`checkComponentAccess` throws `HttpAccessDeniedException` for unauthorized users — the correct webtrees behaviour.

### Test

With the chart component restricted to members, assert each tab AJAX action returns 403 for a guest and 200 for a member.

### LINKED ISSUE #174: Surname joins should use n_num=0, not n_type='NAME' (two 1 NAME lines still fan out) — MarriageMatrix #168 follow-up + NameRepository passdown
### Summary

The primary-name restriction `n_type = 'NAME'` is **not** the full primary-name predicate. webtrees writes one `name` row per top-level `1 NAME` line, **all** carrying `n_type = 'NAME'` with an incrementing `n_num` (`GedcomImportService::updateNames`: `foreach ($record->getAllNames() as $n => $name) { 'n_num' => $n, 'n_type' => $name['type'] }`). An individual with two `1 NAME` lines therefore produces two rows that both pass `n_type = 'NAME'` (with `n_num` 0 and 1). The canonical single-primary-name predicate is `n_num = 0`.

This affects two surname-join sites:

1. **`src/Repository/MarriageMatrixRepository.php:108,114`** — the husband/wife joins (recently changed from the `_MARNM` deny-list to `n_type = 'NAME'` in #168) still fan a marriage into two surname pairs when a spouse has two differently-surnamed `1 NAME` records, because `DISTINCT f_id, h_surn, w_surn` keeps both (the surnames differ). The docblock's "each spouse contributes exactly one surname per family" is still violated for this case. #168 fixed the transliteration/alias case (different `n_type`); this is the remaining two-`1 NAME` case.
2. **`src/Repository/NameRepository.php` `passdownPairsByCentury`** — joins `parent_name`/`child_name` on `n_type = 'NAME'` without `n_num = 0`. A parent or child with two `1 NAME` lines multiplies that parent-child pair in both the numerator (`matches`) and denominator (`total`), skewing the per-century given-name passdown **rate** and pushing `total` across the `MIN_COHORT_SIZE` floor.

### Correct precedent

`GivenNameTrendsRepository` already guards with `n_num = 0`.

### Fix

Add `n_num = 0` to both joins at each site (replacing or supplementing `n_type = 'NAME'`), so each individual contributes exactly its primary name.

### Test

Extend the marriage-matrix fan-out fixture (and add a passdown fixture) with an individual carrying **two top-level `1 NAME` lines** with different surnames — the discriminator a single-`NAME` fixture cannot produce. Assert the marriage is counted once / the passdown rate is unaffected by the second name row.

### Manifestation

Real but uncommon — needs an individual with two top-level `1 NAME` lines (legal name change, dual-culture name, transcription variant entered as a full NAME). Follow-up to #168.

### LINKED ISSUE #175: Performance: reduce whole-tree GEDCOM-blob scans on large trees (averageGenerationLength, occupationInheritance, place-dispersion, residences, migration)
### Summary

Live profiling against a ~2.3k-individual tree plus a code read surface several repository methods that transfer and regex-parse whole-tree GEDCOM blobs and will scale badly on a large tree (100k individuals / 20 generations). These are distinct from the already-fixed #154 (generation-depth N+1 + given-name fold). All proposed fixes reuse helpers the module already ships (`DedupedEventDates::query`, `GedcomScanner::anchoredLikePatterns`, the `?array $cache` memo pattern) and must be behaviour-preserving.

### Findings (ranked)

1. **Critical — `TreeHealthRepository::averageGenerationLength()`** (`src/Repository/TreeHealthRepository.php` ~339-390). A 5-table edge join selects BOTH `parent.i_gedcom` AND `child.i_gedcom` (full blobs) and regex-parses both per parent-child edge. A parent of *k* children appears in *k* rows → their birth-record blob is transferred and parsed *k* times. The only super-linear (edges × fan-out) blob transfer in the module. **Fix:** pull birth years once per individual from the indexed `dates` table via `DedupedEventDates::query($tree, 'BIRT')` (already used elsewhere), then a thin `link → families → link` join selecting only the parent/child xref columns, looking both years up in the in-memory map.

2. **High — `OccupationInheritanceRepository::occupationInheritance()`** (~94-167). Loads every individual's full `i_gedcom` into a resident map keyed by xref for the whole pass. On 100k that is tens–hundreds of MB resident, though most individuals carry no `1 OCCU`. **Fix:** prefilter the SQL to OCCU-bearers via `GedcomScanner::anchoredLikePatterns('OCCU')` (a father without OCCU is dropped anyway).

3. **Medium — `PlaceDispersionRepository`** (`:109` and `:188`). `getMigrationDistanceDistribution()` and `dispersionSummary()` each independently `SELECT i_gedcom FROM individuals` (full-table) and regex every blob; both feed the Places tab. **Fix:** load the blobs once into a per-instance cache and have both methods iterate the shared collection.

4. **Medium — `CountryRepository::residencesByCountry()`** (~118-145). Full-table `SELECT i_gedcom` then `extractAllEventPlaces(…,'RESI')` per individual, even though only RESI-bearers contribute (the sibling `countByCountry()` is already efficient via the `placelinks` join). **Fix:** anchored-LIKE `RESI` prefilter (or join through `placelinks` like the sibling).

5. **Medium — `MigrationRepository::flowsByCountry()`** (~81-135). Full-table blob scan; only individuals with BOTH a birth and death place contribute. **Fix (optional):** restrict to rows whose gedcom contains both `\n1 BIRT` and `\n1 DEAT` via anchored LIKEs before transfer.

### Verification

Behaviour-preserving: each fix must return identical results to the current implementation (pin with the existing integration tests; add a fixture row that exercises the prefilter boundary so the LIKE doesn't drop a contributor). Optionally benchmark each AJAX card before/after against the 105k benchmark tree; #1 should drop from multi-second to sub-100ms and Overview-tab peak PHP memory should fall after #2.

### LINKED ISSUE #176: PHP style: const-before-constructor + parenthesise compound-condition operands
### Summary

Minor PHP code-style deviations from the project's own conventions (not CI-gated by the current cs-fixer ruleset, so they slipped through).

### Findings

1. **`src/Repository/GenerationDepthRepository.php:74`** — a `private const MAX_CHAINS_RENDERED` is declared **after** the constructor. Every other class in the module declares constants before properties/constructor. Move the const block above `__construct`.

2. **Un-parenthesised comparison operands in `&&`/`||`** (the project convention parenthesises comparison/instanceof operands; the codebase is ~98% consistent):
   - `src/Repository/LifeSpanRepository.php:793-794` — `$perSex['M']['n'] < self::MIN_COHORT_SIZE && $perSex['F']['n'] < self::MIN_COHORT_SIZE` (the near-identical block in `ParenthoodRepository` IS parenthesised — direct inconsistency).
   - `src/Repository/LifeSpanRepository.php:1427` — `$band['max'] === null || $age <= $band['max']`.
   - `src/Statistic.php:196` — `$decadeFrom !== null && $decadeTo !== null`.
   - `src/Support/Locale/IsoCountryMap.php:212` — `$name !== '' && $name !== $iso2`.

### Fix

Reorder the const; parenthesise each comparison operand, e.g. `(($perSex['M']['n'] < self::MIN_COHORT_SIZE) && ($perSex['F']['n'] < self::MIN_COHORT_SIZE))`. Pure style, no behaviour change.

### LINKED ISSUE #177: Test gaps: oldestLivingRecord() untested; FamilyRanking N+1 guard uses a loose bound
### Summary

Two test-quality gaps surfaced in a release-readiness review of the suite (otherwise exceptionally well-disciplined: 695 unit + 287 integration, green).

### Findings

1. **`LifeSpanRepository::oldestLivingRecord()` is untested** (medium). It is not a thin delegation — it carries real walk-until-plausible logic: skip `!birthDate->isOK()`, skip `birthJd <= 0`, skip `years <= 0`, and a plausibility cap `years > maxAge` (an implausibly old BIRT year must not win the slot), returning the first plausible candidate or `null`. Its deceased sibling `oldestDeceasedRecord()` IS directly tested; the living counterpart is the asymmetric gap. Add an integration test on a fixture with (a) a living individual with an implausible BIRT year that must be skipped by the `maxAge` cap, (b) a living individual with an unparseable/year-less BIRT skipped by `isOK`, and (c) the genuine oldest-plausible living individual that must be returned — so each `continue` branch is a real discriminator.

2. **`FamilyRankingRepositoryIntegrationTest` N+1 guard uses a loose absolute bound** (low, T17). It asserts `assertLessThanOrEqual((6 * count($result)) + 2, $queryCount)` on a single fixture — wide slack that a partially-restored per-family re-count can slip under, and an incidental upstream query-count shift flakes. Mirror the `GenerationDepthRepositoryIntegrationTest::summaryResolvesChainWithoutPerMemberQueries` pattern instead: import a small and a large grandchild-families fixture of the same shape and assert `assertLessThanOrEqual($smallQueryCount, $largeQueryCount)` plus the existing behavioural assertions — locking the constant-cost contract without slack.

### Note

`ParentMapRepository::build()`'s memoization short-circuit (`if ($this->cache !== null) return …`) is also untested (low/optional) — a focused test calling `build()` twice and asserting zero queries on the second call would pin the memo contract.

### LINKED ISSUE #178: CSS: translucent sticky nav strip rounds its bottom corners — sharp untinted notch when content scrolls behind
### Summary

The sticky tab-navigation strip (`.wt-stat-nav`, `resources/css/statistics.css:~493`) is translucent (`background: color-mix(in srgb, var(--paper) 92%, transparent)` + `backdrop-filter: blur(12px)`) and rounds its **bottom** corners (`border-radius: 0 0 var(--card-radius) var(--card-radius)`). Once the strip is stuck (`position: sticky`), the card column scrolls up behind its bottom edge, but the translucent fill + backdrop blur are clipped to the rounded bottom arc — leaving a sharp, untinted triangle in each bottom corner where content shows through unblurred.

### Fix

Square the bottom corners of the translucent stuck strip (`border-radius: 0`) and convey depth via the already-present `box-shadow: var(--shadow-sticky)`. To keep the "hero + strip as one rounded-foot unit" look at rest, move the rounded bottom corners onto a non-translucent wrapper sibling and butt the sticky strip flush against it (square the shared + content-facing edges). Contrast: `overflow:hidden` does not clip a `box-shadow`, but `border-radius` does clip the translucent fill/backdrop.

### Verification

Visual — scroll a long tab so cards pass behind the stuck nav strip; confirm the bottom-corner notch before/after squaring the radius (jsdom cannot lay out the translucent clip).

### PR DESCRIPTION
**Pull request by magicsunday**
Fixes for the seven findings from the second full-module verification sweep
(end-user UI + performance profiling + developer reviewer fleet). One atomic
commit per issue, test-first where behaviour changes.

### Privacy / access control
- **`GH-172`** — Hall-of-Fame record tiles leaked a living holder's derived value
  (age / spouse- / child-count) and record URL even though `fullName()`
  privatised the name. Gate all six `RecordRowMapper` factories on the holder's
  `canShow()` — one rendering chokepoint covers every record type and every
  repository that resolves a holder. Closes #172.
- **`GH-173`** — the six AJAX tab actions skipped `Auth::checkComponentAccess`, so
  a visitor could call a tab endpoint directly and bypass a members-only
  restriction on the chart component. Moved the gate into the shared
  `renderTab()`. Closes #173.

### Correctness
- **`GH-174`** — `n_type = 'NAME'` is not the full primary-name predicate (two
  `1 NAME` lines both carry it with differing `n_num`), so MarriageMatrix and
  the given-name passdown still fanned out a spouse/parent with two name lines.
  Pin both to `n_num = 0` (the GivenNameTrends precedent). Completes #168.
  Closes #174.

### Performance (large-tree scaling, behaviour-preserving)
- **`GH-175`** — cut whole-tree GEDCOM-blob scans: `averageGenerationLength`
  parses each birth blob once (xref-only edge join) instead of per parent-child
  edge; `occupationInheritance` / `residencesByCountry` / `migration flows`
  anchor-LIKE the relevant tag-bearers before transfer; `PlaceDispersion`
  memoises its one blob load across both metrics. Existing integration suites
  pin the results unchanged. Closes #175.

### Polish
- **`GH-176`** — const-before-ctor + parenthesised compound conditions. Closes #176.
- **`GH-177`** — cover the previously-untested `oldestLivingRecord()` (alive-age
  cap + oldest-plausible selection) and replace the FamilyRanking N+1 guard's
  loose bound with a sparse-vs-dense scaling-independence assertion. Closes #177.
- **`GH-178`** — square the translucent sticky-nav corners so content scrolling
  behind no longer shows a sharp untinted notch. Closes #178.

A final commit re-anchors the `.po` source-line references the edits shifted
(no msgid/msgstr changes).

---
Full `composer ci:test` green on every commit (hook-enforced); each repository's
integration suite verified individually for the behaviour-preserving perf work;
UI re-verified in the browser (all tabs render, popovers work, no console
errors).

### CODE REVIEW

File: src/Repository/TreeHealthRepository.php [hunk 1]
```diff
@@ -335,13 +335,42 @@ public function missingEventGaps(): array
...
      */
     public function averageGenerationLength(): ?float
     {
-        $rows = DB::table('link AS parent_link')
+        $birthYearByXref = [];
+
+        foreach (
+            TreeScope::table($this->tree, 'individuals')
+                ->select(['i_id', 'i_gedcom'])
+                ->get() as $individual
```
[the remark this answers is not in the document]
Comment (magicsunday) [reply src/Repository/TreeHealthRepository.php]: Good call — fixed in 75ac160. Switched averageGenerationLength (and the two sibling iterate-and-discard scans, residencesByCountry + migration flowsByCountry) from `->get()` to `->cursor()`, so rows stream one at a time and only the extracted year / place tally / resolved sample is retained — no whole-tree blob set resident. occupationInheritance keeps `->get()` because it indexes the blobs for the father lookup, and PlaceDispersion keeps it because the memoised collection is replayed across both metrics.

File: src/Repository/OccupationInheritanceRepository.php [hunk 2]
```diff
@@ -17,6 +17,7 @@
 // only the imports from clone detection so the two method bodies stay covered.
 // jscpd:ignore-start
 use Fisharebest\Webtrees\Tree;
+use Illuminate\Database\Query\Builder;
 use MagicSunday\Webtrees\Statistic\Model\Sankey\SankeyFlowsPayload;
 use MagicSunday\Webtrees\Statistic\Model\Sankey\SankeySample;
 use MagicSunday\Webtrees\Statistic\Support\Database\TreeScope;
```
[the remark this answers is not in the document]
Comment (magicsunday) [reply src/Repository/OccupationInheritanceRepository.php:23]: Not an issue — `GedcomScanner` is already imported in `src/Repository/OccupationInheritanceRepository.php` (the pre-existing code already calls `GedcomScanner::extractFirstTagValue(...)`); this change only adds a second static call to the same already-imported class. `composer ci:test:php:phpstan` and `ci:test:php:lint` pass on this commit, both of which fail on an undefined class reference — so there is no missing import or runtime error.

File: src/Repository/CountryRepository.php [hunk 3]
```diff
@@ -12,6 +12,7 @@
 namespace MagicSunday\Webtrees\Statistic\Repository;
 
 use Fisharebest\Webtrees\Tree;
+use Illuminate\Database\Query\Builder;
 use Illuminate\Database\Query\JoinClause;
 use MagicSunday\Webtrees\Statistic\Support\Aggregator\TopNAggregator;
 use MagicSunday\Webtrees\Statistic\Support\Database\TreeScope;
```
[the remark this answers is not in the document]
Comment (magicsunday) [reply src/Repository/CountryRepository.php:18]: Not an issue — `GedcomScanner` is already imported in `src/Repository/CountryRepository.php` (the pre-existing code already calls `GedcomScanner::extractAllEventPlaces(...)`); this change only adds a second static call to the same already-imported class. `composer ci:test:php:phpstan` and `ci:test:php:lint` pass on this commit, both of which fail on an undefined class reference — so there is no missing import or runtime error.

File: src/Repository/MigrationRepository.php [hunk 4]
```diff
@@ -12,6 +12,7 @@
 namespace MagicSunday\Webtrees\Statistic\Repository;
 
 use Fisharebest\Webtrees\Tree;
+use Illuminate\Database\Query\Builder;
 use MagicSunday\Webtrees\Statistic\Model\Sankey\SankeyFlowsPayload;
 use MagicSunday\Webtrees\Statistic\Model\Sankey\SankeySample;
 use MagicSunday\Webtrees\Statistic\Support\Database\TreeScope;
```
[the remark this answers is not in the document]
Comment (magicsunday) [reply src/Repository/MigrationRepository.php:18]: Not an issue — `GedcomScanner` is already imported in `src/Repository/MigrationRepository.php` (the pre-existing code already calls `GedcomScanner::extractEventPlace(...)`); this change only adds a second static call to the same already-imported class. `composer ci:test:php:phpstan` and `ci:test:php:lint` pass on this commit, both of which fail on an undefined class reference — so there is no missing import or runtime error.

File: src/Repository/TreeHealthRepository.php [hunk 5]
```diff
@@ -335,13 +335,45 @@ public function missingEventGaps(): array
...
+        $birthYearByXref = [];
+
+        // Stream the rows with a cursor: only the extracted year (an int) is
+        // retained per individual, so a large tree never holds all the BIRT
+        // blobs resident at once.
+        foreach (
+            TreeScope::table($this->tree, 'individuals')
+                ->select(['i_id', 'i_gedcom'])
+                ->cursor() as $individual
+        ) {
```
[the remark this answers is not in the document]
Comment (magicsunday) [reply src/Repository/TreeHealthRepository.php:369]: Applied in the latest commit — added the `BIRT` anchored-LIKE prefilter to the birth-year scan, matching the sibling scans. Behaviour-preserving: a birthless individual yields no year and was already dropped at the map's null guard.
